src/content/subgames.cpp

```diff
@@ -37,6 +37,21 @@ bool getGameMinetestConfig(const std::string &game_path, Settings &conf)
 	return conf.readConfigFile(conf_path.c_str());
 }
 
+bool getGameConfig(const std::string &game_path, Settings &conf)
+{
+	std::string conf_path = game_path + DIR_DELIM + "game.conf";
+	return conf.readConfigFile(conf_path.c_str());
+}
+
+std::string getGameName(const std::string &game_path)
+{
+	Settings conf;
+	if(!getGameConfig(game_path, conf) && !conf.exists("name"))
+		return "";
```

Comment:
this fails if there is no conf and name in conf, whereas the rest of the code fallbacks to the folder name